### ISSUE TITLE
Add Scheduler component

### ISSUE THREAD

**Issue by oliviertassinari** [mui-org/material-ui-x#1407]
## Summary 💡

Provide a component to organize an agenda

## Examples 🌈

<img width="667" alt="Capture d’écran 2020-09-04 à 02 00 57" src="https://user-images.githubusercontent.com/3165635/92184850-81c32400-ee52-11ea-8c6e-4123c3835838.png">

## Motivation 🔦

This component is relatively frequently required in enterprise applications.

## Benchmark

https://trello.com/c/aTmsByjW/1707-big-calendar-component

**Comment by shoxter** [comment]
Really looking forward to this! We just purchased Kendo React for now, but given our love for the MUI Component API's, we'd switch to this in a heartbeat if it was comparable or better!

**Comment by Misiu** [comment]
Take a look at https://www.bryntum.com/products/scheduler/.
I've used it years ago with ExtJS, but from the beginning it wast one of the best on the market.
They also offer https://www.bryntum.com/products/calendar/, but I'm not sure if this issue is about Scheduler or Calendar, so I'm linking both.

**Comment by pauldvu** [comment]
Currently using Full Calendar, but would love the flexibility of using an MUI based scheduler!

**Comment by acorbin-vdm** [comment]
Currently using another component to do that, but really really interrested by this implementation directly on mui!
What we are looking for in the Timeline view mode (multi rows to organize per person/per room etc.).
That's required on many use cases on enterprises!

**Comment by woodreamz** [comment]
We are using mui since few years. Since you started to add mui-x, we are replacing our third libraries by mui-x. We already use datagrid, charts, treeview and date pickers. I am impatient to have the Scheduler which is a really important component for us.

**Comment by DawsonJBailey** [comment]
Please make this we need it lol

**Comment by troyschneringer** [comment]
We love the MUI/MUIX components.  We use the Devextreme Reactive charts and scheduler extensively, which recently announced was moving into maintenance mode (https://github.com/DevExpress/devextreme-reactive/blob/master/README.md).  We'd love to convert more to MUIX and would be interested in beta testing anything that may come available.

**Comment by oliviertassinari** [comment]
Interesting retrospective that they provide https://github.com/DevExpress/devextreme-reactive/blob/master/README.md#why-devextreme-reactive-is-now-in-maintenance-support-mode

> DevExtreme Reactive began as an alternative to the main DevExtreme product line with the following objectives:
>
> - Provide fully native components for Angular, React, and Vue. Reduce maintenance costs due to a shared codebase.
> - Offer users of popular open-source UI suites such as Bootstrap, Vuetify, and MUI a set of rich components (DataGrid, Scheduler, Charts).
> - Introduce more flexibility via a comprehensive API and built-in plugin system.
> 
> Since the original release of DevExtreme Reactive, we reached the following conclusions:
>
> - Of all our product suites, only the React line (with MUI integration) generated user interest. That market trend forced us to terminate our Reactive Angular and Reactive Vue produce lines.
> - Due to the popularity of DevExtreme Reactive components for MUI, MUI chose to launch their own set of UI components.
> - Based on research, we discovered that developers preferred out-of-the-box solutions versus writing highly customizable code manually. Accordingly, most opted to use React UI components shipping in our main DevExtreme product library.
>
> At present, we must maintain two competitive product lines in parallel - using resources that could be better spent delivering additional value for our primary JS/TS product line (js.devexpress.com/react).

Back in 2019, after seeing https://github.com/mui/material-ui/pull/18035 I tried to reach out to DevExpress CEO to find some ways to join force but I couldn't get hold on him.

**Comment by TheOneTheOnlyJJ** [comment]
When will work on this be started? Can an initial version be expected this year?

**Comment by flaviendelangle** [comment]
Closed by #17691 

The public alpha should start in a few month, we will make a proper announcement when it's ready 👌

**Comment by oliviertassinari** [comment]
From the perspective of a user, there are no public API they can use yet. It's not in alpha or in PR preview (usually, we close issues as soon as there is a PR preview), so "open" feels the most accurate/clearest. Reopened.

**Comment by kiren-james-radus28** [comment]
Hi All, Is there an update on when we can expect a release? We are currently using a 3rd party library but hoping to move to the MUI scheduler when it is released.

**Comment by babang1da** [comment]
please add

**Comment by jvskriubakken** [comment]
I would really like to test the current version of your scheduler and provide feedback and found bugs - if wanted at this stage.

**Comment by flaviendelangle** [comment]
We are currently preparing the initial alpha launch of the components.
I can't communicate an exact launch date but it should be in the next couple of months 👍 

It's awesome to see that there is a lot of traction around those components and we hope the first version will solve your problems!

**Comment by jvskriubakken** [comment]
Yes, we are making some new stuff - where the scheduler will be of great use! So I'm gonna be the first one to use it as soon you are going to include it in the alpha release.

**Comment by ngelotte** [comment]
Excited to try out the Alpha/beta when it is released. I have a new item that I am developing soon that could really use this and it would be nice to stay in the MUI family.

**Comment by rita-codes** [comment]
Hi everyone 👋

Update on this one: the **Scheduler** is now in **alpha** 🎉
👉 Docs & getting started: https://mui.com/x/react-scheduler/

It includes `EventCalendar` (the big-calendar use case discussed in this issue) and `EventTimeline` for resource-row views. For those of you currently using a 3rd-party scheduler and considering a migration, this is the moment your feedback can still shape the API.

To help us prioritize what to build next, we're also collecting use-case feedback via a short form (2 min, 6 questions):
👉 https://forms.gle/Ksbc91D3PcMiiK5x9

If you try it and hit anything weird, please open an issue with the `scope: scheduler` label so we can triage it quickly. 

Thank u so much 🙇‍♀️
